ml_genn/ml_genn/compilers/event_prop_compiler.py

```diff
@@ -1484,22 +1501,52 @@ def _build_in_hid_neuron_model(self, pop: Population,
             all_reset_vars = additional_reset_vars + genn_model.reset_vars
             logger.debug(f"\tReset variables: {all_reset_vars}")
             compile_state.add_neuron_reset_vars(
-                pop, all_reset_vars, True, False)
+                pop, all_reset_vars, True, dyn_ts_reset_needed)
 
             # Generate ring-buffer write code
+            write_code_timestep = "\n".join(f"tsRing{v}[tsRingOffset + tsRingWriteOffset] = {v};"
+                                            for v in saved_vars_timestep)
+
+            if dyn_ts_reset_needed:
+                write_code_timestep += f"""
+                tsRingWriteOffset++;
+                """
+                # add read and write pointer for timestep-wise ring buffers
+                genn_model.add_var("tsRingWriteOffset", "int", 0, reset=False)
+                genn_model.add_var("tsRingReadOffset", "int", self.example_timesteps, reset=False)
+                read_pointer_code= """
+                    tsRingReadOffset--;
+                """
```

Comment:
no need for the extra whitespace - you're adding it into a block with newlines
```suggestion
                read_pointer_code= "tsRingReadOffset--;"
```